src/main/scala/leon/termination/StructuralSize.scala

```diff
@@ -66,10 +88,30 @@ trait StructuralSize {
       case TupleType(argTypes) => argTypes.zipWithIndex.map({
         case (_, index) => size(tupleSelect(expr, index + 1, true))
       }).foldLeft[Expr](InfiniteIntegerLiteral(0))(Plus)
+      case IntegerType =>
+        FunctionInvocation(typedAbsBigIntFun, Seq(expr)) 
       case _ => InfiniteIntegerLiteral(0)
     }
   }
 
+  def lexicographicDecreasing(s1: Seq[Expr], s2: Seq[Expr], strict: Boolean, sizeOfOneExpr: Expr => Expr): Expr = {
+    // Note: The Equal and GreaterThan ASTs work for both BigInt and Bitvector
+    
+    val sameSizeExprs = for ((arg1, arg2) <- (s1 zip s2)) yield Equals(sizeOfOneExpr(arg1), sizeOfOneExpr(arg2))
+    
+    val greaterBecauseGreaterAtFirstDifferentPos =
+      orJoin(for (firstDifferent <- 0 until scala.math.min(s1.length, s2.length)) yield and(
+          andJoin(sameSizeExprs.take(firstDifferent)),
+          GreaterThan(sizeOfOneExpr(s1(firstDifferent)), sizeOfOneExpr(s2(firstDifferent)))
+      ))
+      
+    if (s1.length > s2.length || (s1.length == s2.length && !strict)) {
+      or(andJoin(sameSizeExprs), greaterBecauseGreaterAtFirstDifferentPos)
```

Comment:
Awesome work!

I'm not sure I understand the argument here about strict decreasing when `s1.length > s2.length`. I do believe this works for the relation processor because we are guaranteed that `greaterBecauseGreaterAtFirstDifferentPos` will be proved for some call but if we wanted to use `lexicographicDecreasing` outside of this processor (eg. in the chain processor) we could run afoul of this assumption. Does this optimization actually help for any the current examples?

Other than this minor gripe, I think we're pretty much good to merge! :)